wodles/aws/tests/test_sqs_message_processor.py

```diff
@@ -0,0 +1,180 @@
+                    "Body": json.dumps(
+                        {
+                            "detail": {
+                                "object": {"key": "key1"},
+                                "bucket": {"name": "name1"},
+                            }
+                        }
+                    ),
+                },
+                {"ReceiptHandle": "h2", "Body": json.dumps({"example": "some_value"})},
+            ],
+            [
+                {"route": {"log_path": "key1", "bucket_path": "name1"}, "handle": "h1"},
+                {"raw_message": {"example": "some_value"}, "handle": "h2"},
+            ],
+        ),
+        (
+            sqs.AWSSSecLakeMessageProcessor,
+            [
+                {
+                    "ReceiptHandle": "h1",
+                    "Body": json.dumps(
+                        {
+                            "detail": {
+                                "object": {"key": "key1"},
+                                "bucket": {"name": "name1"},
+                            }
+                        }
+                    ),
+                },
+                {
+                    "ReceiptHandle": "h2",
+                    "Body": json.dumps(
+                        {
+                            "detail": {
+                                "object": {"key": "key2"},
+                                "bucket": {"name": "name2"},
+                            }
+                        }
+                    ),
+                },
+            ],
+            [
+                {"route": {"log_path": "key1", "bucket_path": "name1"}, "handle": "h1"},
+                {"route": {"log_path": "key2", "bucket_path": "name2"}, "handle": "h2"},
+            ],
+        ),
+        (
+            sqs.AWSS3MessageProcessor,
+            [
+                {
+                    "ReceiptHandle": "h1",
+                    "Body": json.dumps(
+                        {
+                            "Records": [
+                                {
+                                    "s3": {
+                                        "object": {"key": "key1"},
+                                        "bucket": {"name": "name1"},
+                                    }
+                                }
+                            ]
+                        }
+                    ),
+                },
+                {"ReceiptHandle": "h2", "Body": json.dumps({"example": "some_value"})},
+            ],
+            [
+                {"route": {"log_path": "key1", "bucket_path": "name1"}, "handle": "h1"},
+                {"raw_message": {"example": "some_value"}, "handle": "h2"},
+            ],
+        ),
+    ],
+)
+def test_extract_message_info(sqs_processor, sqs_messages, expected):
+    """
+    Test extract_message_info method of AWSQueueMessageProcessor subclasses.
+    """
+    processor = sqs_processor()
+    result = processor.extract_message_info(sqs_messages)
+    assert result == expected
```

Comment:
I would prefer to do the `json.dumps` inside the function. Something like

```python
def test_extract_message_info(sqs_processor, sqs_messages, expected):
    """
    Test extract_message_info method of AWSQueueMessageProcessor subclasses.
    """
    processor = sqs_processor()
    for i, message in sqs_messages:
        sqs_messages[i]['Body'] = json.dumps(message['Body'])
    result = processor.extract_message_info(sqs_messages)
    assert result == expected
```

Reply:
https://github.com/wazuh/wazuh/blob/8f8f66ff33b8f07634b7dfe296be74d9ef01d703/wodles/aws/tests/test_sqs_message_processor.py#L165-L173